File: services/timetracking/entity/timetracking.go
package entity

import (
	"time"
)

type TimeTracking struct {
	ID          string    `json:"id,omitempty"          bson:"_id"`
	CharacterID string    `json:"characterID,omitempty" bson:"character_id"`
	CategoryID  string    `json:"categoryID,omitempty"  bson:"category_id,omitempty"`
	StartTime   time.Time `json:"startTime,omitempty"   bson:"start_time"`
	EndTime     time.Time `json:"endTime,omitempty"     bson:"end_time,omitempty"`
}

type TimeTrackingWithFish struct {
	TimeTracking *TimeTracking `json:"timeTracking"`
	Normal       int           `json:"normal"`
	Gold         int           `json:"gold"`
}

File: services/penk/src/services/grpc/proto/timetracking/TimeTracking.ts
// Original file: ../../proto/timetracking/timetracking_message.proto

import type { Long } from '@grpc/proto-loader';

export interface TimeTracking {
  'id'?: (string);
  'characterId'?: (string);
  'categoryId'?: (string);
  'startTime'?: (number | string | Long);
  'endTime'?: (number | string | Long);
  '_categoryId'?: "categoryId";
  '_endTime'?: "endTime";
}

export interface TimeTracking__Output {
  'id': (string);
  'characterId': (string);
  'categoryId'?: (string);
  'startTime': (string);
  'endTime'?: (string);
  '_categoryId': "categoryId";
  '_endTime': "endTime";
}
